src/plone/rest/traverse.py

```diff
@@ -18,7 +18,10 @@ def publishTraverse(self, request, name):
             obj = super(RESTTraverse, self).publishTraverse(request, name)
             if (not IContentish.providedBy(obj)
                     and not IService.providedBy(obj)):
-                raise KeyError
+                if name in ('virtual_hosting',):
```

Comment:
+1 for the `isinstance` check (as the very first check, like in your original approach in 3b29ac6). @tomgross could you please squash your commits into a single one afterwards?